core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -357,7 +357,14 @@ private void internalMove(String name, Move move) {
    */
   @Override
   public Schema apply() {
-    return applyChanges(schema, deletes, updates, adds, moves);
+    Schema newSchema = applyChanges(schema, deletes, updates, adds, moves);
+
+    // Validate the metrics if we have existing properties.
+    if (base != null && base.properties() != null) {
+      MetricsConfig.fromProperties(base.properties()).validateReferencedColumns(newSchema);
+    }
+
+    return newSchema;
```

Comment:
I don't think that this path is tested. Could you add one?

Reply:
It is currently covered by `testModificationWithMetricsMetrics`, commenting out the check results in the following error when running gradle build:

```
org.apache.iceberg.TestSchemaAndMappingUpdate > testModificationWithMetricsMetrics[formatVersion = 1] FAILED
    java.lang.AssertionError: No exception was thrown (Deleting a column with metrics fails), expected: org.apache.iceberg.exceptions.ValidationException
        at org.junit.Assert.fail(Assert.java:88)
        at org.apache.iceberg.AssertHelpers.assertThrows(AssertHelpers.java:65)
        at org.apache.iceberg.TestSchemaAndMappingUpdate.testModificationWithMetricsMetrics(TestSchemaAndMappingUpdate.java:179)

org.apache.iceberg.TestSchemaAndMappingUpdate > testModificationWithMetricsMetrics[formatVersion = 2] FAILED
    java.lang.AssertionError: No exception was thrown (Deleting a column with metrics fails), expected: org.apache.iceberg.exceptions.ValidationException
        at org.junit.Assert.fail(Assert.java:88)
        at org.apache.iceberg.AssertHelpers.assertThrows(AssertHelpers.java:65)
        at org.apache.iceberg.TestSchemaAndMappingUpdate.testModificationWithMetricsMetrics(TestSchemaAndMappingUpdate.java:179)

```